pkg/index/tree.go

```diff
@@ -319,7 +319,9 @@ func (n *Node[V]) SetChildren(children []*Node[V]) error {
 	n.children = children
 	for _, child := range children {
 		child.Parent = n
-		child.UpdateAncestorsSize()
+		if !child.Value.IsRemoved() {
+			child.UpdateAncestorsSize()
+		}
```

Comment:
This change does not exist in the PR of JS SDK. Could you explain?
https://github.com/yorkie-team/yorkie-js-sdk/pull/756

Reply:
In js-sdk, crdtTreeNode inherits indexTreeNode. That's why deepcopy in [js-sdk can calculate the size](https://github.com/yorkie-team/yorkie-js-sdk/blob/2af8a0c791667f7143eb72599a5c8c4a9f9ecd76/src/document/crdt/tree.ts#L451) right away.
However in the case of go-sdk, there is no such structure, so it seemed necessary to update the size (Length in go-sdk) of the indexTreeNode for ancestors during the process of `setChildren()`.

Reply:
```go
func (n *TreeNode) DeepCopy() (*TreeNode, error) {
        // ...
	clone.RemovedAt = n.RemovedAt
	clone.Index.Length = n.Index.Length
	clone.InsPrevID = n.InsPrevID
	clone.InsNextID = n.InsNextID

	if n.IsText() {
		return clone, nil
	}
        // ...
}

// SetChildren sets the children of the given node.
func (n *Node[V]) SetChildren(children []*Node[V]) error {
	if n.IsText() {
		return ErrInvalidMethodCallForTextNode
	}

	n.children = children
	for _, child := range children {
		child.Parent = n
		// erase here:
		// child.UpdateAncestorsSize() 
	}

	return nil
}
```

I thought of another way to solve it by directly referencing the IndexTreeNode and changing the Length to be copied as well, and not changing the Length in `setChildren()`. What do you think?